dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs

```diff
@@ -139,6 +140,10 @@ public sealed class FunctionCallingStepwisePlanner
                     {
                         chatHistoryForSteps.AddUserMessage($"Failed to execute function {functionResponse.FullyQualifiedName}. Try something else!");
                     }
+                    catch (Exception ex) when (!ex.IsCriticalException())
+                    {
+                        chatHistoryForSteps.AddUserMessage($"Execution of function {functionResponse.FullyQualifiedName} failed with error: \"{ex.Message}\" Try something else!");
```

Comment:
Need to add tool message here, per this error: `An assistant message with 'tool_calls' must be followed by tool messages responding to each 'tool_call_id'.`